### PR TITLE
#164198507 Users should be able to report an article

### PR DESCRIPTION
**Pull request by uniqueayo1988**
#### What does this PR do?
Have CRUD on reports endpoint

#### Description of Task to be completed?
- Create report table
- Implement a POST /api/reports/:slug endpoint
- Implement a GET /api/reports
- Implement a GET /api/reports/:id
- Implement a PATCH /api/reports/:id/status

#### How should this be manually tested?
1. Clone this branch.
2. Run `npm install` to install the dependencies.
3. Run `npm test` to run the tests.

#### Any background context you want to provide?
- A user should be able to report an article other from the ones created by user
- Admin should be able to see reports and update report status

#### What are the relevant pivotal tracker stories?
[#164198507](https://pivotaltracker.com/story/show/#164198507)

### CODE REVIEW

File: server/database/seeders/20190309084305-admin-user.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      isAdmin: false,
+      createdAt: new Date(),
+      updatedAt: new Date()
+    },
+    {
+      firstname: 'bayo',
+      lastname: 'bayo',
+      username: 'bayo',
+      email: 'uniqueayo@yahoo.com',
+      password: '$2a$08$G/5b9Oq3rla2pjb/imcZ8u.m0EQlbKfG.gwGJG6iOfPVm9rZqNVmC',
```
Comment (andela-gike) [inline server/database/seeders/20190309084305-admin-user.js]: Is there a reason why you need to add the hashed password here directly? kindly confirm this.

Comment (andela-gike) [reply server/database/seeders/20190309084305-admin-user.js]: You could call the hash function then passing the password; What do you think?

**Review by andela-gike — APPROVED** [review]

**Review by andela-gike — CHANGES_REQUESTED** [review]
https://github.com/andela/artemis-ah-backend/pull/46#discussion_r264250802

File: server/database/seeders/20190309084305-admin-user.js (hunk 1 above)
Comment (uniqueayo1988) [reply server/database/seeders/20190309084305-admin-user.js]: I needed the hashed password to log in and run my tests. However, it is more efficient to call the hash function as pointed out.
Thank you for pointing this out. I'll do the needful.

**Review by andela-gike — APPROVED** [review]

**Pull request merged**